File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/products/dto/ProductResponse.java
package com.portfolio.auctionmarket.domain.products.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.portfolio.auctionmarket.domain.products.entity.Product;
import com.portfolio.auctionmarket.domain.products.entity.ProductImage;
import com.portfolio.auctionmarket.domain.products.entity.ProductStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductResponse {

    private Long productId;
    private String seller;
    private String category;
    private String title;
    private ProductStatus productStatus;
    private String mainImageUrl;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdAt;

    public static ProductResponse from(Product entity) {
        String mainUrl = entity.getImage().stream()
                .filter(img -> img.getImageOrder().equals(1))
                .map(ProductImage::getImageUrl)
                .findFirst()
                .orElse("https://picsum.photos/400/300");

        return ProductResponse.builder()
                .productId(entity.getProductId())
                .seller(entity.getSeller().getNickname())
                .category(entity.getCategory().getCategory())
                .title(entity.getTitle())
                .productStatus(entity.getProductStatus())
                .mainImageUrl(mainUrl)
                .createdAt(entity.getCreatedAt())
                .build();
    }
}

File: AuctionMarket/auction-domain/src/main/kotlin/org/example/auctioncommon/domain/product/dto/ProductResponse.kt
package org.example.auctioncommon.domain.product.dto

import com.fasterxml.jackson.annotation.JsonFormat
import org.example.auctioncommon.domain.product.entity.Product
import org.example.auctioncommon.domain.product.entity.ProductStatus
import java.time.LocalDateTime

data class ProductResponse(
    val productId: Long,
    val seller: String,
    val category: String,
    val title: String,
    val productStatus: ProductStatus,
    val mainImageUrl: String,

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val createdAt: LocalDateTime
)

fun Product.toProductResponse(): ProductResponse {

    val mainUrl = this.image
        .firstOrNull { it.imageOrder == 1 }
        ?.imageUrl ?: "https://picsum.photos/400/300"

    return ProductResponse(
        productId = this.productId ?: 0L,
        seller = this.seller.nickname,
        category = this.category.category,
        title = this.title,
        productStatus = this.productStatus,
        mainImageUrl = mainUrl,
        createdAt = this.createdAt ?: LocalDateTime.now()
    )
}
